pkg/controller/operators/olm/operator.go

```diff
@@ -989,20 +987,26 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 
-		// Check if any generated resources are missing
-		if resErr := a.checkAPIServiceResources(out, certs.PEMSHA256); len(resErr) > 0 {
-			// Check if API services are adoptable. If not, keep CSV as Failed state
-			if a.apiServiceResourceErrorsActionable(resErr) {
-				out.SetPhaseWithEvent(v1alpha1.CSVPhasePending, v1alpha1.CSVReasonAPIServiceResourcesNeedReinstall, resErr[0].Error(), now, a.recorder)
-				return
+		// Check if any generated resources are missing and that OLM can action on them
```

Comment:
nit: replace and with so

Reply:
What I'm trying to say here is that we're checking two things:
1. Are resources missing?
2. Can OLM perform an action based on the errors returned?

Maybe something like "Check for any actionable APIResource issues" fits better.